### PR TITLE
Fix A6 crash, improve aesthetics, disallow DB queries on main thread

### PR DESCRIPTION
**Pull request by pulsejet**
Fixes crash on Marshmallow among other things

### CODE REVIEW

File: app/src/main/res/layout/fragment_feed.xml [hunk 1]
```diff
@@ -13,8 +13,7 @@
         <android.support.v7.widget.RecyclerView
             android:id="@+id/feed_recycler_view"
             android:layout_width="match_parent"
-            android:layout_height="match_parent"
-            android:paddingTop="8dp" />
```
Comment (unstablebrainiac) [inline app/src/main/res/layout/fragment_feed.xml]: Why?

File: app/src/main/res/layout/fragment_news.xml [hunk 2]
```diff
@@ -13,8 +13,7 @@
         <android.support.v7.widget.RecyclerView
             android:id="@+id/news_recycler_view"
             android:layout_width="match_parent"
-            android:layout_height="match_parent"
-            android:paddingTop="8dp" />
```
Comment (unstablebrainiac) [inline app/src/main/res/layout/fragment_news.xml]: Why?

File: app/src/main/res/layout/fragment_placement_blog.xml [hunk 3]
```diff
@@ -13,8 +13,7 @@
         <android.support.v7.widget.RecyclerView
             android:id="@+id/placement_feed_recycler_view"
             android:layout_width="match_parent"
-            android:layout_height="match_parent"
-            android:paddingTop="8dp" />
```
Comment (unstablebrainiac) [inline app/src/main/res/layout/fragment_placement_blog.xml]: Why?

File: app/src/main/res/layout/fragment_training_blog.xml [hunk 4]
```diff
@@ -13,8 +13,7 @@
         <android.support.v7.widget.RecyclerView
             android:id="@+id/training_feed_recycler_view"
             android:layout_width="match_parent"
-            android:layout_height="match_parent"
-            android:paddingTop="8dp" />
```
Comment (unstablebrainiac) [inline app/src/main/res/layout/fragment_training_blog.xml]: Why?

File: app/src/main/res/layout/fragment_feed.xml (hunk 1 above)
Comment (pulsejet) [reply app/src/main/res/layout/fragment_feed.xml]: That's the ugly gap on top (same for others)

**Pull request merged**